waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -118,16 +182,25 @@ constructAppSpec waspDir options packageJson parsedPrismaSchema decls = do
 
   return $ runValidation ASV.validateAppSpec appSpec
 
-findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
+data WaspFile
+  = WaspLangFile {_path :: !(Path' Abs File')}
+  | WaspTsFile {_path :: !(Path' Abs File')}
+  deriving (Show)
```

Comment:
Interesting, feels a bit unusual. I get why you did it though, you want to be able to say it is one of them but you don't want to yet know which one. Records in sum types are usually an anti-pattern, but these records are identical except for the data constructor, so it works.

What if you want to specify somewhere that a function needs specifically path to WaspTS file? Or Sepcifically WaspLangFile? We have that knowledge but we didn't put it in StrongPath.

I guess soltuion would be
```hs
data WaspLangFile
data WaspTsFile
data WaspFile = WaspLang (Path' Abs WaspLangFile) | WaspTs (Path' Abs WaspTsFile)
```
And I wouldn't bother with `_path`, I probably won't be needing that anyway as I will most likely always want to deconstruct hm.
But ok, this is verbose a bit, and it is all quite localized so I guess it is not important. Although you do export `WaspFile (..)`, which makes it less localized hm. Ok, I am ok with this what you have, although I would be tempted to go this extra step.